origin.spec

```diff
@@ -0,0 +1,428 @@
+# builders provided in the rel-eng directory of this project
+Version:        0.0.1
+Release:        0%{?dist}
+Summary:        Open Source Container Management by Red Hat
+License:        ASL 2.0
+URL:            https://%{import_path}
+ExclusiveArch:  x86_64
+Source0:        https://%{import_path}/archive/%{commit}/%{name}-%{version}.tar.gz
+BuildRequires:  systemd
+BuildRequires:  golang >= 1.4
+Obsoletes: openshift >= 1.0
+Obsoletes: openshift <= 3.1
+
+
+%description
+%{summary}
+
+%package master
+Summary:        %{product_name} Master
+Requires:       %{name} = %{version}-%{release}
+Requires(post): systemd
+Requires(preun): systemd
+Requires(postun): systemd
+
+%description master
+%{summary}
+
+%package node
+Summary:        %{product_name} Node
+Requires:       %{name} = %{version}-%{release}
+Requires:       docker-io >= %{docker_version}
+Requires:       tuned-profiles-%{name}-node
+Requires:       util-linux
+Requires:       socat
+Requires:       nfs-utils
+Requires(post): systemd
+Requires(preun): systemd
+Requires(postun): systemd
+
+%description node
+%{summary}
+
+%package -n tuned-profiles-%{name}-node
+Summary:        Tuned profiles for %{product_name} Node hosts
+Requires:       tuned >= %{tuned_version}
+Requires:       %{name} = %{version}-%{release}
+
+%description -n tuned-profiles-%{name}-node
+%{summary}
+
+%package clients
+Summary:      %{product_name} Client binaries for Linux, Mac OSX, and Windows
+BuildRequires: golang-pkg-darwin-amd64
+BuildRequires: golang-pkg-windows-386
+
+%description clients
+%{summary}
+
+%package dockerregistry
+Summary:        Docker Registry v2 for %{product_name}
+Requires:       %{name} = %{version}-%{release}
+
+%description dockerregistry
+%{summary}
+
+%package pod
+Summary:        %{product_name} Pod
+Requires:       %{name} = %{version}-%{release}
+
+%description pod
+%{summary}
+
+%package sdn-ovs
+Summary:          %{product_name} SDN Plugin for Open vSwitch
+Requires:         openvswitch >= %{openvswitch_version}
+Requires:         %{name}-node = %{version}-%{release}
+Requires:         bridge-utils
+Requires:         ethtool
+
+%description sdn-ovs
+%{summary}
+
+%prep
+%setup -q
+
+%build
+
+# Don't judge me for this ... it's so bad.
+mkdir _build
+
+# Horrid hack because golang loves to just bundle everything
+pushd _build
+    mkdir -p src/github.com/openshift
+    ln -s $(dirs +1 -l) src/%{import_path}
+popd
+
+
+# Gaming the GOPATH to include the third party bundled libs at build
+# time. This is bad and I feel bad.
+mkdir _thirdpartyhacks
+pushd _thirdpartyhacks
+    ln -s \
+        $(dirs +1 -l)/Godeps/_workspace/src/ \
+            src
+popd
+export GOPATH=$(pwd)/_build:$(pwd)/_thirdpartyhacks:%{buildroot}%{gopath}:%{gopath}
+# Build all linux components we care about
+for cmd in openshift dockerregistry
+do
+        go install -ldflags "%{ldflags}" %{import_path}/cmd/${cmd}
+done
+
+# Build clients for other platforms
+# TODO: build cmd/oc instead of openshift
```

Comment:
Ha, sorry, I should have tagged @sdodson.  I'm just wondering if I should at this task to the card for shipping a client subpackage.